runtime/server/server.go

```diff
@@ -146,80 +162,37 @@ func (s *Server) ServeGRPC(ctx context.Context) error {
 	runtimev1.RegisterRuntimeServiceServer(server, s)
 	runtimev1.RegisterQueryServiceServer(server, s)
 	runtimev1.RegisterConnectorServiceServer(server, s)
-	s.logger.Sugar().Infof("serving runtime gRPC on port:%v", s.opts.GRPCPort)
-	return graceful.ServeGRPC(ctx, server, s.opts.GRPCPort)
-}
 
-// Starts the HTTP server.
-func (s *Server) ServeHTTP(ctx context.Context, registerAdditionalHandlers func(mux *http.ServeMux)) error {
-	handler, err := s.HTTPHandler(ctx, registerAdditionalHandlers)
+	transcoder, err := vanguardgrpc.NewTranscoder(server)
 	if err != nil {
-		return err
-	}
-
-	server := &http.Server{Handler: handler}
-	s.logger.Sugar().Infof("serving HTTP on port:%v", s.opts.HTTPPort)
-	options := graceful.ServeOptions{
-		Port:     s.opts.HTTPPort,
-		CertPath: s.opts.TLSCertPath,
-		KeyPath:  s.opts.TLSKeyPath,
-	}
-	return graceful.ServeHTTP(ctx, server, options)
-}
-
-// HTTPHandler HTTP handler serving REST gateway.
-func (s *Server) HTTPHandler(ctx context.Context, registerAdditionalHandlers func(mux *http.ServeMux)) (http.Handler, error) {
-	// Create REST gateway
-	gwMux := gateway.NewServeMux(
-		gateway.WithErrorHandler(HTTPErrorHandler),
-		gateway.WithOutgoingHeaderMatcher(func(s string) (string, bool) {
-			// grpc gateway adds gateway.MetadataHeaderPrefix to all outgoing headers
-			// we want to skip that for `x-trace-id` set in response
-			if s == observability.TracingHeader {
-				return s, true
-			}
-			// default matcher logic
-			return fmt.Sprintf("%s%s", gateway.MetadataHeaderPrefix, s), true
```

Comment:
Does the new implementation propagate `x-trace-id` correctly? Or do we need to restore this middleware somehow?

Reply:
It looks like it is successfully propagating the initial header via `LoggingUnaryServerInterceptor`